### ISSUE TITLE
Ignore images within `data-pagefind-ignore`

### ISSUE THREAD

**Issue by vanbroup** [CloudCannon/pagefind#19]
Currently there is no ability to exclude images selected for the image meta data, this is mainly problematic when there is no image related to the content at which an image from the site design is selected instead. 

I tried to exclude these images by putting them in a `data-pagefind-ignore` container expecting those images to be ignored but without success.

As alternative I'm overruling the image meta data with the automatically generated open graph image, but this prevents any other images from within the content to be selected.

```html
<meta property="og:image" content="{{- $img.Permalink -}}" data-pagefind-meta="image[content]"/>
```

**Comment by bglw** [comment]
Hello! 👋 

Hmm yes I see. It is intentional that `data-pagefind-ignore` only affects the indexing — for example the `<head>` is an implicit `data-pagefind-ignore`, but we still want to pluck filters and metadata out of it. In another case, I have sites where I wrap a sidebar navigation in `data-pagefind-ignore`, but some data inside that sidebar is useful for setting up a filter group. 

I might look at expanding that attribute to take an argument. Then the current argumentless version would imply `data-pagefind-ignore="index"`, and we could add a `data-pagefind-ignore="all"` which also omits filters and metadata.

How does that sound to you?

**Comment by vanbroup** [comment]
That would work, thanks.

**Comment by vanbroup** [comment]
Another thought here is that you can fall back to the open graph image when the content search has not provided any results, I don't think that this is something that can currently be configured.

**Comment by bglw** [comment]
Yes good call. If you're defining your own `data-pagefind-meta` then subsequent definitions will overwrite the earlier ones, but there isn't a configuration to have the auto-detected image take precedence. 

One way to allow this would be adding some configuration to the Pagefind UI along the lines of
```js
{
  image: ["image", "og-image"]
}
```

Where the UI tries to use those keys in order for the image. Which would let the default image stay as `meta.image`, and you could separately tag your open graph image as `og-image`. Then at the UI end it can fall back to the `og-image` if no auto image was found.

**Comment by vanbroup** [comment]
That could be a good approach, this meta data would also need to be ignored in the `skipMeta` in that case.

**Comment by bglw** [comment]
Hi @vanbroup  👋 

Some fixes have landed in [Pagefind v0.5.0](https://github.com/CloudCannon/pagefind/releases/tag/v0.5.0) 🎉 

- `data-pagefind-ignore="all"` has been implemented as described above
- An alternate solution for fallback content was implemented, allowing you to tag your og image as `data-pagefind-default-meta`
